internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixStoreGraveyard.java

```diff
@@ -77,9 +77,35 @@ public int getLargestUsedVersionNumber(String storeName) {
     return largestUsedVersionNumber;
   }
 
+  @Override
+  public int getLargestUsedRTVersionNumber(String storeName) {
+    if (VeniceSystemStoreUtils.isSystemStore(storeName)) {
+      VeniceSystemStoreType systemStoreType = VeniceSystemStoreType.getSystemStoreType(storeName);
+      if (systemStoreType != null && systemStoreType.isStoreZkShared()) {
+        String userStoreName = systemStoreType.extractRegularStoreName(storeName);
+        return getPerUserStoreSystemStoreLargestUsedRTVersionNumber(userStoreName, systemStoreType);
```

Comment:
Why do we need special handling for system stores? Can we try to avoid code duplication?